apps/web/src/pages/Teams/TeamMembersPage/TeamMembersPage.tsx

```diff
@@ -1,11 +1,108 @@
 import * as React from "react";
-import { useIntl } from "react-intl";
-import { BuildingOffice2Icon } from "@heroicons/react/24/outline";
+import { IntlShape, useIntl } from "react-intl";
+import { useParams } from "react-router-dom";
+import orderBy from "lodash/orderBy";
+
+import { Heading, Pending, Pill, ThrowNotFound } from "@gc-digital-talent/ui";
+import { getLocalizedName } from "@gc-digital-talent/i18n";
+import { notEmpty } from "@gc-digital-talent/helpers";
 
-import PageHeader from "~/components/PageHeader";
 import SEO from "~/components/SEO/SEO";
+import Table, { ColumnsOf, Cell } from "~/components/Table/ClientManagedTable";
+import {
+  Maybe,
+  Role,
+  Scalars,
+  Team,
+  useAllUsersQuery,
+  useGetTeamQuery,
+  useListRolesQuery,
+  User,
+} from "~/api/generated";
+import { getFullNameLabel } from "~/utils/nameUtils";
+import { groupRoleAssignmentsByUser, TeamMember } from "~/utils/teamUtils";
 
-const TeamMembersPage = () => {
+import AddTeamMemberDialog from "./components/AddTeamMemberDialog";
+import EditTeamMemberDialog from "./components/EditTeamMemberDialog";
+import RemoveTeamMemberDialog from "./components/RemoveTeamMemberDialog";
+
+const orderRoles = (roles: Array<Role>, intl: IntlShape) => {
+  return orderBy(roles, ({ displayName }) => {
+    const value = getLocalizedName(displayName, intl);
+
+    return value
+      ? value
+          .normalize("NFD")
+          .replace(/[\u0300-\u036f]/g, "")
+          .toLocaleLowerCase()
+      : value;
+  });
+};
+
+const actionCell = (user: TeamMember, team: Team, roles: Array<Role>) => (
+  <div
+    data-h2-display="base(flex)"
+    data-h2-flex-wrap="base(wrap)"
+    data-h2-gap="base(x.25)"
+  >
+    <EditTeamMemberDialog user={user} team={team} availableRoles={roles} />
+    <RemoveTeamMemberDialog user={user} team={team} />
+  </div>
+);
+
+const emailLinkCell = (email: Maybe<string>, intl: IntlShape) => {
+  if (email) {
+    return <a href={`mailto:${email}`}>{email}</a>;
+  }
+
+  return (
+    <span data-h2-font-style="base(italic)">
+      {intl.formatMessage({
+        defaultMessage: "No email provided",
+        id: "1JCjTP",
+        description: "Fallback for email value",
+      })}
+    </span>
+  );
+};
+
+const roleCell = (roles: Maybe<Maybe<Role>[]>, intl: IntlShape) => {
+  const nonEmptyRoles = roles?.filter(notEmpty);
+  const rolePills = nonEmptyRoles
+    ? orderRoles(nonEmptyRoles, intl).map((role) => (
+        <Pill color="neutral" mode="solid" key={role.id}>
+          {getLocalizedName(role.displayName, intl)}
+        </Pill>
+      ))
+    : null;
+
+  return rolePills ? (
+    <span data-h2-display="base(flex)" data-h2-gap="base(x.25)">
+      {rolePills}
+    </span>
+  ) : null;
+};
+
+const roleAccessor = (roles: Maybe<Maybe<Role>[]>, intl: IntlShape) => {
+  const nonEmptyRoles = roles?.filter(notEmpty);
+
+  return orderBy(nonEmptyRoles, intl).join(", ");
+};
```

Comment:
not sure how the other comments got submitted already, I wasn't done yet 😕 

anyway, searching by roles doesn't seem to work
You've got `orderBy()` called here, but that doesnt need intl, the `orderRoles()` you've defines above does. Mix-up?

Reply:
https://recordit.co/nxoGtAVDil